security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java

```diff
@@ -82,7 +83,16 @@ private static char[] generateMasterPassword() {
             String password = UUID.randomUUID().toString();
             try {
                 Files.writeString(path, password, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW);
-                Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
+                if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
+                    Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
+                } else {
+                    File file = path.toFile();
+                    boolean readable = file.setReadable(true, true);
+                    boolean executable = file.setExecutable(true, true);
+                    if (!readable || !executable) {
+                        LOGGER.log(Level.DEBUG, "Could not set file permission to " + file.toPath());
```

Comment:
the whole thing should be guarded by `LOGGER.isLoggable` 
also `file.toPath()` does not seem to make sense here, as we already have a `path` variable containing the result.